internal/util/k8s/secrets.go

```diff
@@ -143,9 +134,15 @@ func (sc *secretCache) updateCache(secret *corev1.Secret, force bool) map[string
 		data: secretData,
 	}
 
+	// We should cache this secret only if it was requested in the first place i.e force == true
+	// For all subsequent updates where this function is called, it is set to false
 	sc.Lock()
 	defer sc.Unlock()
 
+	if _, requested := sc.cache[cKey]; !requested && !force {
+		return nil
+	}
+
```

Comment:
```suggestion
	// We should cache this secret only if it was already present in the first place i.e force == true
	// For all subsequent updates where this function is called, it is set to false
	sc.Lock()
	defer sc.Unlock()

	if _, present := sc.cache[cKey]; !found && !force {
		return nil
	}

```
I was confused by the use of `requested`, can we instead use `present`  ?

Reply:
If it helps, it will only be present if it was requested by us in the first place. And in that step we are just checking that. 

We ignore events for all other secrets.